packages/manager/apps/dedicated/client/app/billing/main/payments/billing-payments.service.js

```diff
@@ -1,158 +1,72 @@
-          cache: billingCache,
-          serviceType: 'apiv7',
-        })
-        .then((response) => {
-          const res = response.data.sort(
-            (a, b) => ids.indexOf(a.key) - ids.indexOf(b.key),
-          );
-
-          return res.map((item) =>
-            item.error
-              ? angular.extend({ error: item.error }, item.value)
-              : item.value,
-          );
-        });
-    };
-
-    this.getPayment = (id) =>
-      $http
-        .get(`/me/deposit/${id}`, {
-          cache: billingCache,
-          serviceType: 'apiv7',
-        })
-        .then((response) => response.data);
-
-    this.getBillsIds = (id) =>
-      $http
-        .get(`/me/deposit/${id}/paidBills`, {
-          cache: billingCache,
-          serviceType: 'apiv7',
-        })
-        .then((response) => response.data);
-
-    this.getBillDetails = (id, billId) =>
-      this.getBill(id, billId).then((bill) =>
-        this.getOperationsDetails(id, billId, bill.orderId).then(
-          (operations) => {
-            const billdetails = clone(bill);
-            if (!operations || operations.length === 0) {
-              throw new Error('No operation for a bill concerned by a deposit');
-            }
-            if (operations.length > 1) {
-              throw new Error(
-                'More than one operation for a bill concerned by a deposit',
-              );
-            }
-            billdetails.payment = operations[0].amount.text;
-            return billdetails;
+    }
+
+    getBills(id, { offset, pageSize }) {
+      return this.iceberg(`/me/deposit/${id}/paidBills`)
+        .query()
+        .expand('CachedObjectList-Pages')
+        .offset(offset || 1)
+        .limit(pageSize)
+        .execute()
+        .$promise.then((response) => ({
+          data: response.data,
+          meta: {
+            totalCount:
+              parseInt(response.headers['x-pagination-elements'], 10) || 0,
           },
-        ),
-      );
-
-    this.getBill = (id, billId) =>
-      $http
-        .get(`/me/deposit/${id}/paidBills/${billId}`, {
-          cache: billingCache,
-          serviceType: 'apiv7',
-        })
-        .then((response) => response.data);
-
-    this.getOperationsDetails = (id, billId, orderId) =>
-      this.getOperationsIds(id, billId, orderId).then((operationsIds) =>
-        $q.all(
-          map(operationsIds, (operationId) =>
-            this.getOperation(id, billId, operationId),
-          ),
-        ),
+        }));
+    }
+
+    getBillDetails(id, bill) {
+      return this.getOperationsDetails(id, bill.billId, bill.orderId).then(
+        (operations) => {
+          if (!operations || operations.length === 0) {
+            throw new Error('No operation for a bill concerned by a deposit');
+          }
+          if (operations.length > 1) {
+            throw new Error(
+              'More than one operation for a bill concerned by a deposit',
+            );
+          }
```

Comment:
Do we catch or do something with those error ?

Reply:
Will check. Not in the scope of this PR. But if I can do something, I'll do it ;)